lab_4_summarization_textrank/README.md

```diff
@@ -0,0 +1,758 @@
+
+Методы `set_attribute` (`set_preprocessed`, `set_encoded`) принимают на вход соответствующее значение и 
+присваивают его соответствующему атрибуту экземпляра класса.
+
+**Важно**: должна происходить проверка элементов на соответствие типа:
+* для `set_preprocessed` - то, что все элементы строки
+* для `set_encoded` - то, что все элементы числа
+
+Если это условие не выполняется, то должна подниматься ошибка `ValueError`.
+
+Пример интерфейса:
+```py
+def set_encoded(self, encoded_sentence: EncodedSentence):
+    pass
+```  
+
+Методы `get_attribute` (`get_text`, `get_preprocessed`, `get_encoded`, `get_position`) не принимают ничего на вход и возвращают текущее 
+значение соответствующего атрибута.
+
+Пример интерфейса:
+```py
+def get_encoded(self) -> EncodedSentence:
+    pass
+```  
+
+### Шаг 3. Выделить из текста предложения
+
+В рамках [лабораторной работы №3](../lab_3_keywords_textrank/README.md) Вы реализовали класс `TextPreprocessor`, позволяющий предобрабатывать текст.
+В данной работе Вы модифицируете функционал этого класса с помощью наследования и переопределения методов, чтобы можно было работать с предложениями.
+
+В этом шаге Вы будете реализовывать `SentencePreprocessor`.
+
+#### Шаг 3.1 Инициализатор класса
+
+Для создания объекта `SentencePreprocessor` нам необходимо знать: список стоп слов (аргумент `stop_words`) и пунктуационные знаки (аргумент `punctuation`).
+
+Внутри инициализатора необходимо вызвать инициализатор родительского класса с пришедшими аргументами.
+
+Таким образом, у объекта класса `SentencePreprocessor` должны быть как минимум следующие поля: `_stop_words`, `_punctuation`.
+
+Интерфейс:   
+```py
+class SentencePreprocessor(TextPreprocessor):
+    def __init__(self, stop_words: tuple[str, ...], punctuation: tuple[str, ...]):
+        pass
+```  
+
+**Важно**: должна происходить проверка элементов на соответствие типа:
+* для `stop_words` - то, что все элементы строки
+* для `punctuation` - то, что все элементы строки
+
+Если это условие не выполняется, то должна подниматься ошибка `ValueError`.
+
+
+#### Шаг 3.2 Разделение текста на предложение. Метод `_split_by_sentence`
+
+Этот метод, как вытекает из названия, разделяет текст на отдельные предложения. 
+
+Метод принимает на вход неочищенный текст. 
+
+Возвращаемым значением метода является кортеж предложений.
+
+**Важно**: каждый элемент возвращаемого кортежа - экземпляр класса `Sentence`.
+
+**Важно №2**: критерий разделения текста на предложения:
+* Знак препинания (`.!?`), за которым следует заглавная буква.
+
+Интерфейс:   
+```py
+def _split_by_sentence(self, text: str) -> tuple[Sentence, ...]:
+    pass
+```  
+
+#### Шаг 3.3 Предобработка предложений. Метод `_preprocess_sentences`
+
+Этот метод предобрабатывает (разделяет на отдельные токены и очищает от специальных символов) предложения. 
+
+Метод принимает на вход последовательность предложений. 
+
+Метод добавляет к каждому предложению его предобработанную версию.
```

Comment:
looks like it is time to give an example and/(or at least) explain connection with set_preprocessed